tests/main/broken-seeding/task.yaml

```
@@ -7,10 +7,8 @@ environment:
     SEED_DIR: /var/lib/snapd/seed
 
 prepare: |
-    if [ "$TRUST_TEST_KEYS" = "false" ]; then
-        echo "This test needs test keys to be trusted"
-        exit
-    fi
+    echo "This test needs test keys to be trusted"
+    SKIP [ "$TRUST_TEST_KEYS" = "false" ]
```

Personally I'd prefer this:
```suggestion
    [ "$TRUST_TEST_KEYS" = "false" ] && SKIP
```

As this way the shell test is not magic in any way and the real work is just executing a program.
In addition this would allow for the SKIP to take an argument explaining why we skip the test.